website_snippet_carousel_product/controllers/website.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2020 Tecnativa - Alexandre Díaz
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import time
+
+from odoo import http
+from odoo.http import request
+
+from odoo.addons.website.controllers.main import QueryURL
+
+
+class Website(http.Controller):
+    @http.route(
+        ["/website/render_product_carousel"],
+        type="json",
+        auth="public",
+        website=True,
+        csrf=False,
+        cache=30,
+    )
+    def render_product_carousel(
+        self, template, domain=False, limit=12, products_per_slide=4, **kwargs
```

Comment:
Why passing the template as an argument? I think we should it that for avoiding exploits or information leak.